env-var-docs/parser-package/src/env_var_docs/parser.py

```diff
@@ -0,0 +1,524 @@
+"""Provides a 'visit' function that parses an environment variable
+documentation file and returns any parsing errors due to unexpected structure.
+If no parsing errors are found, 'visit' runs a user-provided function on each
+node in the file.
+
+See parser-package/README.md for a description of the expected structure.
+"""
+
+import dataclasses
+import typing
+import json
+
+UnparsedJSON = dict[str, typing.Any]
+
+
+@dataclasses.dataclass
+class Group:
+    group_description: str
+    members: dict[str, UnparsedJSON]
+
+
+@dataclasses.dataclass
+class RegexTest:
```

Comment:
It is a test case of a Variable's regex field.  In the env-var-docs.json file, if someone adds a regex validation check, they also need to add regex_tests.  This makes it much less likely that regex typos or errors are submitted to env-var-docs.json.

I agree it could be a bit weird to see a Test class in a non-test python file but I think RegexTest is the most clear name for the struct

Reply:
Added.